### PR TITLE
ci: bump nixpkgs versions

### PR DESCRIPTION
**Pull request by ipetkov**
* Update to a more recent revision of nixpkgs
* Add nixpkgs-21.11 for testing
* Add nixpkgs-21.05 for testing
* Drop nixpkgs-20.09, branch is EOL
* Drop nixpkgs-20.05, branch is EOL
* Bump ripgrep-all to v0.9.6 to make tests pass

### CODE REVIEW

File: nix/sources.nix [hunk 1]
```diff
@@ -98,7 +98,10 @@ let
       saneName = stringAsChars (c: if isNull (builtins.match "[a-zA-Z0-9]" c) then "_" else c) name;
       ersatz = builtins.getEnv "NIV_OVERRIDE_${saneName}";
     in
-      if ersatz == "" then drv else ersatz;
+      if ersatz == "" then drv else
```
Comment (nmattia) [inline nix/sources.nix:101]: What was the issue here? Would you like to create a ticket in https://github.com/nmattia/niv?

**Review by nmattia — APPROVED** [review]
Thanks a lot!

File: nix/sources.nix (hunk 1 above)
Comment (ipetkov) [reply nix/sources.nix:101]: Ah I should have mentioned... This was volunteered by niv itself (IIRC bumping the source version from 23 to 24).

Diff comes from invoking `niv init` and I decided to keep it for the sake of being on the latest-and-greatest version of everything :)

Comment (nmattia) [reply nix/sources.nix:101]: hah, I see! Thanks

**Pull request merged**